kernel/src/process.rs

```diff
@@ -771,6 +785,15 @@ impl Process<'a> {
             && buf_end_addr <= self.mem_break()
     }
 
+    crate fn allow(&self, buf_start_addr: *const u8, size: usize) -> Option<AppSlice<Shared, u8>> {
```

Comment:
Perhaps Result be more fitting here, given that it's treated that way by the caller anyway (and deals with invalid input)?

Reply:
Done, and also moved the logic to differentiate between a null and non-null allow call to here, so the semantics are centralized.